### PR TITLE
IBX-2964: Flatpickr widget localization

### PR DESCRIPTION
**Pull request by barw4**
| Question      | Answer
| ------------- | ---
| Tickets       | [IBX-2964](https://issues.ibexa.co/browse/IBX-2964)
| Bug fix?      | no
| New feature?  | no
| BC breaks?    | no
| Tests pass?   | yes
| Doc needed?   | no
| License       | [GPL-2.0](https://github.com/ezsystems/ezplatform-admin-ui/blob/master/LICENSE)

The language will be picked up from the User's Back Office settings.

#### Checklist:
- [x] Coding standards (`$ composer fix-cs`)
- [x] Ready for Code Review

### CODE REVIEW

**Comment by konradoboza** [comment]
It seems the JIRA issue link is wrong.

File: src/bundle/Resources/public/js/scripts/widgets/flatpickr.js [hunk 1]
```diff
@@ -0,0 +1,8 @@
+import flatpickrLanguages
+    from '../../../../../../../../../../public/bundles/ezplatformadminuiassets/vendors/flatpickr/dist/l10n';
+
+(function (eZ, flatpickr) {
+    const backOfficeLanguage = eZ.adminUiConfig.backOfficeLanguage;
+    const flatpickrLanguage = flatpickrLanguages[backOfficeLanguage] || flatpickrLanguages.default;
+    flatpickr.localize(flatpickrLanguage);
```
Comment (GrabowskiM) [inline src/bundle/Resources/public/js/scripts/widgets/flatpickr.js:8]:
```suggestion

    flatpickr.localize(flatpickrLanguage);

```

File: src/bundle/Resources/public/js/scripts/widgets/flatpickr.js [hunk 2]
```diff
@@ -0,0 +1,8 @@
+import flatpickrLanguages
+    from '../../../../../../../../../../public/bundles/ezplatformadminuiassets/vendors/flatpickr/dist/l10n';
+
+(function (eZ, flatpickr) {
+    const backOfficeLanguage = eZ.adminUiConfig.backOfficeLanguage;
+    const flatpickrLanguage = flatpickrLanguages[backOfficeLanguage] || flatpickrLanguages.default;
```
Comment (GrabowskiM) [inline src/bundle/Resources/public/js/scripts/widgets/flatpickr.js]:
```suggestion
    const flatpickrLanguage = flatpickrLanguages[backOfficeLanguage] ?? flatpickrLanguages.default;

```

File: src/bundle/Resources/public/js/scripts/widgets/flatpickr.js [hunk 3]
```diff
@@ -0,0 +1,8 @@
+import flatpickrLanguages
+    from '../../../../../../../../../../public/bundles/ezplatformadminuiassets/vendors/flatpickr/dist/l10n';
+
+(function (eZ, flatpickr) {
+    const backOfficeLanguage = eZ.adminUiConfig.backOfficeLanguage;
```
Comment (GrabowskiM) [inline src/bundle/Resources/public/js/scripts/widgets/flatpickr.js]:
```suggestion
    const { backOfficeLanguage } = eZ.adminUiConfig;

```

File: src/bundle/Resources/public/js/scripts/widgets/flatpickr.js [hunk 4]
```diff
@@ -0,0 +1,8 @@
+import flatpickrLanguages
+    from '../../../../../../../../../../public/bundles/ezplatformadminuiassets/vendors/flatpickr/dist/l10n';
+
+(function (eZ, flatpickr) {
```
Comment (GrabowskiM) [inline src/bundle/Resources/public/js/scripts/widgets/flatpickr.js]:
```suggestion
(function (global, doc, eZ, flatpickr) {
```
these two parameters (global and doc) should be always included, even if they're not used at the moment, as for consistency and readability they should be as first two args, and we shouldn't change order of params in the future... I think. @dew326

**Review by GrabowskiM — CHANGES_REQUESTED** [review]

File: src/bundle/Resources/public/js/scripts/widgets/flatpickr.js [hunk 5]
```diff
@@ -0,0 +1,10 @@
+import flatpickrLanguages
+    from '../../../../../../../../../../public/bundles/ezplatformadminuiassets/vendors/flatpickr/dist/l10n';
+
+(function (global, doc, eZ, flatpickr) {
+    const { backOfficeLanguage } = eZ.adminUiConfig;
+    const flatpickrLanguage = flatpickrLanguages[backOfficeLanguage] ?? flatpickrLanguages.default;
+
+    flatpickr.localize(flatpickrLanguage);
+
```
Comment (GrabowskiM) [inline src/bundle/Resources/public/js/scripts/widgets/flatpickr.js]:
```suggestion
```
Sorry, my mistake in previous suggestion, I added one new line too much

File: src/bundle/Resources/public/js/scripts/widgets/flatpickr.js [hunk 6]
```diff
@@ -0,0 +1,10 @@
+import flatpickrLanguages
+    from '../../../../../../../../../../public/bundles/ezplatformadminuiassets/vendors/flatpickr/dist/l10n';
+
+(function (global, doc, eZ, flatpickr) {
+    const { backOfficeLanguage } = eZ.adminUiConfig;
+    const flatpickrLanguage = flatpickrLanguages[backOfficeLanguage] ?? flatpickrLanguages.default;
+
+    flatpickr.localize(flatpickrLanguage);
+
+})(global, doc, window.eZ, window.flatpickr);
```
Comment (GrabowskiM) [inline src/bundle/Resources/public/js/scripts/widgets/flatpickr.js]:
```suggestion
})(window, window.document, window.eZ, window.flatpickr);

```

**Review by GrabowskiM — CHANGES_REQUESTED** [review]

**Review by OstafinL — APPROVED** [review]

File: src/lib/UI/Config/Provider/BackOfficeLanguage.php [hunk 7]
```diff
@@ -0,0 +1,36 @@
...
+ * @license For full copyright and license information view LICENSE file distributed with this source code.
+ */
+declare(strict_types=1);
+
+namespace Ibexa\AdminUi\UI\Config\Provider;
+
+use EzSystems\EzPlatformAdminUi\UI\Config\ProviderInterface;
+use EzSystems\EzPlatformUser\UserSetting\UserSettingService;
+
+class BackOfficeLanguage implements ProviderInterface
```
Comment (konradoboza) [inline src/lib/UI/Config/Provider/BackOfficeLanguage.php]:
```suggestion
final class BackOfficeLanguage implements ProviderInterface
```

**Review by konradoboza — APPROVED** [review]

**Review by GrabowskiM — APPROVED** [review]

**Review by mikadamczyk — APPROVED** [review]

**Review by tomaszszopinski — APPROVED** [review]
QA approved on IbexaDXP 3.3

**Pull request merged**